test/cpp/interop/interop_server.cc

```diff
@@ -152,6 +153,16 @@ class TestServiceImpl : public TestService::Service {
     return Status::OK;
   }
 
+  // Response contains current timestamp. We ignore everything in the request.
+  Status CacheableUnaryCall(ServerContext* context, const SimpleRequest* request,
+                   SimpleResponse* response) {
+    gpr_timespec ts = gpr_now(GPR_CLOCK_REALTIME);
+    std::string timestamp = std::to_string(ts.tv_nsec);
+    response->mutable_payload()->set_body(timestamp.c_str(), timestamp.size());
+    context->AddInitialMetadata("Cache-Control", "max-age=100000, public");
```

Comment:
As implemented, this can cause flakiness, if the cached response is quite old already on the first request and expires for before the second. Maybe reduce the max-age to a minute or similar and have each client specify something unique (maybe random number) in the request? There should also probably be something in the response so that clients can verify the initial request was un-cached, since there could be a bug where all requests to the same URL were cached independent of the request (say, if the Vary header was missing).